### ISSUE TITLE
Rendering performance is bad compared to other terminal emulators

### ISSUE THREAD

**Issue by BuzzwordChief** [wez/wezterm#927]
Benchmarking Wezterm by ```cat```ing a 1.6GB file to stdout revealed how slow Wezterm is compared to other macOS compatible Terminals like Alacritty/Kitty/Terminal.app etc. They all achived the task in under one minute while Wezterm needed +25 minutes (It took so long I thought I used the wrong command but a second run confirmed that the correct command was issued).

 - OS: macOS 10.14.6
 - Version: wezterm 20210502-154244-3f7122cb
 - EurKey (US-ANSI similar)

__Configuration__
Font size is set to 15pt, no other settings were changed. The same font and font size was used in all tests

<img width="1737" alt="grafik" src="https://user-images.githubusercontent.com/11695209/124658699-6080f900-dea4-11eb-8ebd-210c8dd8c9fb.png">

**Comment by wez** [comment]
Please try the nightly build!

**Comment by BuzzwordChief** [comment]
<img width="1501" alt="grafik" src="https://user-images.githubusercontent.com/11695209/124691715-8fb35c80-dedc-11eb-846f-275bc3930484.png">
This time i stopped the benchmark in wezterm after 6 minutes, it seems like their is no difference to the release build.

**Comment by dev-zero** [comment]
@wez while trying to see whether or not a different application is leaking filehandles, I was playing around with `opensnoop`, surprisingly I found that `wezterm` seems to constantly reopen `/lib/terminfo/l/linux` (well, on Linux obviously):

```
~ ❯ doas opensnoop
PID    COMM               FD ERR PATH
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
35284  wezterm-gui        44   0 /lib/terminfo/l/linux
...
```

... maybe this has something to do with the performance?

**Comment by wez** [comment]
@dev-zero That terminfo opening stuff was fixed in d1d7def1a142da7a38da13491f7869f08351d78f and 3ea030794b158e3d02b6cca1e3cc3eb9bd544b1d

**Comment by dev-zero** [comment]
@wez good to know, thanks!
